crmsh/conf_parser.py

```diff
@@ -228,49 +64,76 @@ def get(self, path, index=0):
         path: config path
         index: known index in section
         """
-        path = self._real_path(path, index)
         try:
-            return eval("self._config_inst.{}".format(path))
-        except (AttributeError, IndexError):
+            return self._dom_query.get(path, index)
+        except (KeyError, IndexError):
             return None
 
     def get_all(self, path):
         """
         Returns all values matching path
         """
-        if not self._is_list_path(path):
-            value = self.get(path)
-            return [value] if value else []
-        return [self.get(path, index=i) for i in range(self._len_of_list(path))]
+        return self._dom_query.get_all(path)
 
     def remove(self, path, index=0):
-        """
-        """
-        value = self.get(path, index)
-        if not value:
-            raise ValueError("Cannot find value on path \"{}:{}\"".format(path, index))
-        real_path = self._real_path(path, index)
-        exec("del self._config_inst.{}".format(real_path))
+        try:
+            self._dom_query.remove(path, index)
+        except (KeyError, IndexError):
+            raise ValueError("Cannot find value on path \"{}:{}\"".format(path, index)) from None
+
+    def _raw_set(self, path, value, index):
+        path = path.split('.')
+        node = self._dom
+        path_stack = tuple()
+        for key in path[:-1]:
+            path_stack = (*path_stack, key)
+            if key not in node:
+                new_node = dict()
+                node[key] = new_node
+                node = new_node
+            else:
+                match node[key]:
+                    case dict(_) as next_node:
+                        if index > 0 and path_stack in self._sec_names_with_list:
+                            if index == 1:
+                                new_node = dict()
+                                node[key] = [next_node, new_node]
+                                node = new_node
+                            else:
+                                raise IndexError(f'index out of range: {index}')
```

Comment:
Worth to add a behave test case to cover this exception

And
```suggestion
                                raise IndexError(f'Index out of range: {index}')
```

Reply:
Using lower case in IndexError is consistent with python standard library.